### PR TITLE
eos: Ban violent/adult apps from solutions/spark products

### PR DESCRIPTION
**Pull request by mazen-asef**
https://phabricator.endlessm.com/T27150

### CODE REVIEW

File: plugins/eos/gs-plugin-eos.c [hunk 1]
```diff
@@ -1395,7 +1395,10 @@ app_is_banned_for_personality (GsPlugin *plugin, GsApp *app)
 	       ((g_strcmp0 (priv->personality, "zh_CN") == 0) &&
 	        (g_strv_contains (google_apps, app_name) ||
 	         g_str_has_prefix (app_name, "com.endlessm.encyclopedia"))) ||
-	       (g_str_has_prefix (priv->personality, "spark") &&
+               (g_str_has_prefix (priv->product_name, "spark") &&
```
Comment (pwithnall) [inline plugins/eos/gs-plugin-eos.c]: Nitpick: This line is over-indented by 1 space, which makes the nesting of the `&&`s and `||`s quite a lot harder to understand.

**Comment by mazen-asef** [comment]
@pwithnall I have made some changes on the logic too, to add the hack product as Rob mentioned on a e-mail chain - @ramcq is this acceptable, or you would like to do the hack change as part of another task?

**Comment by pwithnall** [comment]
> This is fine to push to `master` once the nitpick is fixed, but be aware that as part of the gnome-software rebase to GNOME 3.32, this hard-coded blacklist is going to go away. Please make sure that all the apps you want to blacklist have appropriate OARS metadata set — the parental controls system which will replace this is based on that.

To clarify this:
 * Please make sure you add OARS rating data to all these apps **now** (see https://phabricator.endlessm.com/T26920)
 * `app_is_banned_for_personality()` and its related hard-coded blacklist functionality is not actually removed by the current version of the gnome-software rebase
 * However, `app_is_banned_for_personality()` and friends *have* moved from the gnome-software `eos` plugin to the `eos-blacklist` plugin; your patch here should be largely applicable to the moved version

File: plugins/eos/gs-plugin-eos.c [hunk 2]
```diff
@@ -1391,13 +1389,16 @@ app_is_banned_for_personality (GsPlugin *plugin, GsApp *app)
 		return FALSE;
 
 	return ((g_strcmp0 (priv->personality, "es_GT") == 0) &&
-	        g_strv_contains (violent_apps, app_name)) ||
+	        g_strv_contains (restricted_apps, app_name)) ||
```
Comment (pwithnall) [inline plugins/eos/gs-plugin-eos.c]: This change to combine `violent_apps` and `adult_apps` means that adult apps are now blocked on `es_GT`. Is that intentional?

File: plugins/eos/gs-plugin-eos.c [hunk 3]
```diff
@@ -1391,13 +1389,16 @@ app_is_banned_for_personality (GsPlugin *plugin, GsApp *app)
...
 	       ((g_strcmp0 (priv->personality, "zh_CN") == 0) &&
 	        (g_strv_contains (google_apps, app_name) ||
-	         g_str_has_prefix (app_name, "com.endlessm.encyclopedia"))) ||
-	       (g_str_has_prefix (priv->personality, "spark") &&
-	        (g_strv_contains (adult_apps, app_name) ||
-	         g_strv_contains (violent_apps, app_name)));
+	        g_str_has_prefix (app_name, "com.endlessm.encyclopedia"))) ||
+               ((g_strcmp0 (priv->product_name, "hack") == 0) &&
```
Comment (pwithnall) [inline plugins/eos/gs-plugin-eos.c]: Indentation is too deep here.

File: plugins/eos/gs-plugin-eos.c (hunk 2 above)
Comment (ramcq) [reply plugins/eos/gs-plugin-eos.c]: Yeah, the same sensitivities (won't somebody think of the children!!!!) apply, but when we were actively selling in es_GT, we didn't have Flathub so these adult apps didn't exist. There's no product reason to maintain a separate list, it's just the passage of time. (And lack of ability atm to set a platform-wide filter on extreme violence / extreme nudity / ...)

**Comment by mazen-asef** [comment]
So, i checked all apps being blacklisted here on the restricted_apps array and they do have OARS ratings, though `ws.openarena.OpenArena` and `com.grangerhub.Tremulous` have the 1.0 version which shouldn't be a problem for g-s >=3.27 (not sure where we stand now)

File: plugins/eos/gs-plugin-eos.c (hunk 3 above)
Comment (mazen-asef) [reply plugins/eos/gs-plugin-eos.c]: Sorry, I moved the line un-willingly :(

**Comment by pwithnall** [comment]
I fixed a couple of other whitespace problems, squashed the fixup, and merged. Thanks.

**Pull request merged**